src/hooks/useApiConnection.ts

```diff
@@ -78,7 +78,7 @@ export function useApiConnection(connectionDetails: ConnectionChainInformation):
       setConfigs(values);
     };
 
-    if (isReady && isEmpty(configs)) {
+    if (apiPromise.isReady && isEmpty(configs)) {
       getChainConfigs();
     }
   }, [apiPromise, configs, isReady]);
```

Comment:
do we need `isReady` dependency?

Reply:
Yes you do. If `apiPromise.isReady` is removed then on `getConfigs` you will receive on:
```
const systemChain = await apiPromise.rpc.system.name();
```
a `rpc of undefined not found` or something like that (happened to me)

Reply:
We probably need `[apiPromise, apiPromise.isReady, configs]` instead.

Reply:
@wirednkod `isReady` is not used within the `useEffect`, so I feel there is something weird happening here. Like obviously it works, it just indicates something fishy. Why do we check `apiPromise.isReady` and not `isReady`? Shouldn't the two be synchronized anyway?

Reply:
The `isReady` is expecting the `apiPromise.isReady` to be true and the `configs` to be empty in order to retrieve them.
On line `59` (https://github.com/paritytech/parity-bridges-ui/blob/5e8851725cb774981b88ef3c4fea8732978ffede/src/hooks/useApiConnection.ts#L59-L62) there is a similar check but where `isReady` is false. If here we checked the `isReady` we must wait for `apiPromise.isReady` + 1 render to make the `isReady` true, in order to finally get the `configs`. There is no reason to do that - saves us 1 extra render.